src/aggregator/aggregator/aggregator_test.go

```diff
@@ -984,34 +984,44 @@ func TestAggregatorOwnedShards(t *testing.T) {
 	}
 }
 
-func TestAggregatorAddMetricMetrics(t *testing.T) {
+func TestAggregatorAddUntimedMetrics(t *testing.T) {
 	s := tally.NewTestScope("testScope", nil)
 	m := newAggregatorAddUntimedMetrics(s, instrument.TimerOptions{StandardSampleRate: 0.001})
 	m.ReportSuccess()
 	m.SuccessLatencyStopwatch().Stop()
-	m.ReportError(errInvalidMetricType)
-	m.ReportError(errShardNotOwned)
-	m.ReportError(errAggregatorShardNotWriteable)
-	m.ReportError(errWriteNewMetricRateLimitExceeded)
-	m.ReportError(errWriteValueRateLimitExceeded)
-	m.ReportError(xerrors.NewRenamedError(errArrivedTooLate, fmt.Errorf("errorrr")))
-	m.ReportError(errors.New("foo"))
+	for _, state := range []ElectionState{LeaderState, FollowerState} {
+		m.ReportError(errInvalidMetricType, state)
+		m.ReportError(errShardNotOwned, state)
+		m.ReportError(errAggregatorShardNotWriteable, state)
+		m.ReportError(errWriteNewMetricRateLimitExceeded, state)
+		m.ReportError(errWriteValueRateLimitExceeded, state)
+		m.ReportError(xerrors.NewRenamedError(errArrivedTooLate, fmt.Errorf("errorrr")), state)
```

Comment:
Nit - you can get rid of that `fmt` import by using `errors.New` instead:
```suggestion
		m.ReportError(xerrors.NewRenamedError(errArrivedTooLate, errors.New("errorrr")), state)
```

Reply:
https://github.com/m3db/m3/pull/3559/commits/63d14798b89bf8d027377e9ff0c30069df16c2c4